tools/doms/doms_reader.py

```diff
@@ -0,0 +1,133 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from netCDF4 import Dataset
+import sys
+import datetime
+import csv
+from collections import OrderedDict
+
+def assemble_matches(filename):
+    """
+    Read a DOMS netCDF file and return a list of matches.
+    
+    Arguments: 
+        filename (string): the DOMS netCDF file name.
+    
+    Returns:
+        matches (list): List of matches. Each list element is a dictionary:
+            For netCDF group GROUP (SatelliteData or InsituData) and group
+                variable VARIABLE:
+            matches[GROUP]['matchID']: MatchedRecords dimension ID for the match
+            matches[GROUP]['GROUPID']: GROUP dim dimension ID for the record
+            matches[GROUP][VARIABLE]: variable value 
+    """
+    # Open the netCDF file.
+    try:
+        doms_nc = Dataset(filename, 'r')

```

Comment:
I added a doms_nc.close() statement to address this so that I could keep the try/except block. Or would one of [these solutions](https://stackoverflow.com/questions/5627425/what-is-a-good-way-to-handle-exceptions-when-trying-to-read-a-file-in-python) be better?

Reply:
Ah I see. I think probably doing it the way the question asker did in that question is the appropriate way to ensure the resource is always closed properly. Your changes improved it but if an exception is thrown somewhere before `doms_nc.close()` is called the resource is leaked. Which, to be fair, is not a major concern since this is a simple python script and the OS should be able to reclaim the resource when the python process exits. I'd still advocate for wrapping a `with` in the `try`

Reply:
Gotcha. I updated the netCDF file and CSV file handling to be a `with` in a `try`.